### PR TITLE
Use the timezone inside the component instead of passing it 5x

### PR DESCRIPTION
**Pull request by rnbwdsh**

### CODE REVIEW

**Comment by 31453** [comment]
@rnbwdsh Thanks! This is a good idea. The MolochTime component does indeed have a getter for the user object which includes their timezone setting. The same goes for the MolochSearch component.

However, the MolochTime and MolochSearch components also need to be updated to exclude the `timezone` prop and use `user.settings.timezone` instead.

Also, I'm going to mark this PR and issue with the HackTogether label. If you want swag, please [sign-up](https://yahoo.github.io/hacktogether/) for Yahoo! Hack Together, if you have not done so, yet.

**Review by 31453 — CHANGES_REQUESTED** [review]
MolochTime and MolochSearch components also need to be updated to exclude the `timezone` prop and use `user.settings.timezone` instead.

**Comment by rnbwdsh** [comment]
> Good idea, I'll fix that too.
In SessionField, it uses
`this.timezone || this.$store.state.user.settings.timezone,`
Whereas the passed in timezone is from the user.settings anyways, so the whole `this.timezone ||` can be omitted as well as the prop.

In fact I could edit it out in most places without any side-effects.

**Comment by 31453** [comment]
That's a great idea!

**Comment by 31453** [comment]
Also, it looks like the user object from the store is not available at the time that the `datePickerOptions` is being initialized. So, you'll have to wait to set `datePickerOptions` and initialize the time fields until the user object is available.

**Comment by 31453** [comment]
And FYI: it looks like there's an existing bug when changing the user timezone settings. It always sets the timezone to the previously selected setting. I'm created a new issue to reference this: #1667 No need to fix it in this PR unless you want to. I just wanted to make you aware of this issue.

File: viewer/vueapp/src/components/sessions/StickySessions.vue [hunk 1]
```diff
@@ -176,6 +176,9 @@ export default {
       } else {
         return this.sessions;
       }
+    },
+    timezone: function () {
+      return this.$store.state.user.timezone;
```
Comment (31453) [inline viewer/vueapp/src/components/sessions/StickySessions.vue:181]: needs to be `this.$store.state.user.settings.timezone`

File: viewer/vueapp/src/components/visualizations/Visualizations.vue [hunk 2]
```diff
@@ -370,6 +366,9 @@ export default {
           this.$store.commit('updateSeriesType', newValue);
         }
       }
+    },
+    timezone: function () {
+      return this.$store.state.user.timezone;
```
Comment (31453) [inline viewer/vueapp/src/components/visualizations/Visualizations.vue:371]: needs to be `this.$store.state.user.settings.timezone`

**Review by 31453 — CHANGES_REQUESTED** [review]
just a few minor changes with `this.$store.state.user.settings.timezone`

**Comment by rnbwdsh** [comment]
Rebased + Fixed by now, thanks for the great support @31453.

**Review by 31453 — APPROVED** [review]

**Comment by 31453** [comment]
Looks great! Thanks for all the hard work ✨

**Pull request merged**